File: shasi/src/types/patient.ts
export type Gender = 'male' | 'female' | 'other';
export type SkinType = 'normal' | 'dry' | 'oily' | 'combination' | 'sensitive';

export interface Patient {
  id: string;
  patient_code: string;
  full_name: string;
  photo_url: string | null;
  date_of_birth: string | null;
  gender: Gender | null;
  phone: string | null;
  whatsapp: string | null;
  email: string | null;
  address: string | null;
  allergy_history: string | null;
  medical_conditions: string | null;
  skin_type: SkinType | null;
  notes: string | null;
  tags: string[];
  is_active: boolean;
  created_by: string | null;
  created_at: string;
  updated_at: string;
}

export interface PatientFormData {
  full_name: string;
  date_of_birth?: string;
  gender?: Gender;
  phone?: string;
  whatsapp?: string;
  email?: string;
  address?: string;
  allergy_history?: string;
  medical_conditions?: string;
  skin_type?: SkinType;
  notes?: string;
  tags?: string[];
}

File: sc-pos-be/internal/models/patient.go
package models

import "time"

// Patient represents clinic patient
type Patient struct {
	ID                string       `json:"id" db:"id"`
	PatientCode       string       `json:"patient_code" db:"patient_code"`
	FullName          string       `json:"full_name" db:"full_name"`
	PhotoURL          *string      `json:"photo_url" db:"photo_url"`
	DateOfBirth       NullableTime `json:"date_of_birth" db:"date_of_birth"`
	Gender            *string      `json:"gender" db:"gender"` // male, female, other
	Phone             *string      `json:"phone" db:"phone"`
	WhatsApp          *string      `json:"whatsapp" db:"whatsapp"`
	Email             *string      `json:"email" db:"email"`
	Address           *string      `json:"address" db:"address"`
	AllergyHistory    *string      `json:"allergy_history" db:"allergy_history"`
	MedicalConditions *string      `json:"medical_conditions" db:"medical_conditions"`
	SkinType          *string      `json:"skin_type" db:"skin_type"`
	Notes             *string      `json:"notes" db:"notes"`
	Tags              []string     `json:"tags" db:"tags"`
	IsActive          bool         `json:"is_active" db:"is_active"`
	ReminderOptIn     *bool        `json:"reminder_opt_in" db:"reminder_opt_in"`
	CreatedBy         *string      `json:"created_by" db:"created_by"`
	UpdatedBy         *string      `json:"updated_by,omitempty" db:"updated_by"`
	DeletedAt         *time.Time   `json:"deleted_at,omitempty" db:"deleted_at"`
	CreatedAt         time.Time    `json:"created_at" db:"created_at"`
	UpdatedAt         time.Time    `json:"updated_at" db:"updated_at"`
}
